### ISSUE TITLE
общая переменная для шаблонов bemtree

### ISSUE THREAD

**Issue by Guria** [bem/bem-forum-content-ru#309]
Написал такое:
```js
var title = 'my long title';
block('page').elem('title').content()(function(){ return title });
block('kg-appbar').elem('title').content()(function(){ return title });
```
Не сработало, т.к. переменная и шаблоны разъхались в разные области видимости после компиляции.
Есть решение или мне не стоит такого хотеть?

**Comment by verybigman** [comment]
`applyNext({ _title : 'my long title' })` - но помни, что это зло и юзать это надо аккуратно

**Comment by tadatuta** [comment]
@verybigman если переменная по сути глобальная, то я бы не сказал, что это зло, вполне норм.
@Guria к методу, который предложил Антон, добавлю еще про возможность добавлять методы в прототип: https://github.com/bem/bem-core/pull/636#issuecomment-51981145

**Comment by Guria** [comment]
@verybigman немного не понял как именно в описанном случае поможет applyNext(). Я переменную определил за пределами шаблонов.

То что моя хотелка, мягко говоря странная, сам понимаю. Но вопрос интересен с теоретической точкой зрения:
Каким образом получить доступ к общей переменной в двух не связанных между собой bemtree шаблонах.

**Comment by tadatuta** [comment]
@Guria идея в том, что `this` общее вглубь по дереву, так что если положить что-то в `this` в блоке `page`, оно окажется доступно всем остальным шаблонам.

**Comment by Guria** [comment]
Всё понял, спасибо.

**Comment by Guria** [comment]
А при вызове `BEMTREE.apply()` в этот `this` как-то положить что-нибудь можно?
Или только в шаблонах это возможно?

**Comment by dab** [comment]
@Guria 
В BEMTREE.apply() можно передать любые данные:

```js
BEMTREE.apply({
    block: 'island',
    data: customData
})
```

Это отвечает на вопрос?

**Comment by tadatuta** [comment]
@Guria 
> А при вызове `BEMTREE.apply()` в этот `this` как-то положить что-нибудь можно?

Нет, но можно пробрасывать данные шаблоне на `match(functionI() { return !this.data; })`, т.е. на условие, что данные еще не были проброшены или вызывать BEMTREE.apply()` на какой-нибудь корневой блок `root` и передавать данные + тот контекст, который хочется шаблонизировать на самом деле. Я приводил пример какое-то время назад в комментариях к https://ru.bem.info/forum/issues/157/

**Comment by Guria** [comment]
Спасибо за ссылку. Теперь когда я таки добрался до bemtree+bemhtml стало понятно про что там речь шла. Надо ссылку сделать на все твои комменты более 10 строк на форуме :)

**Comment by zxqfox** [comment]
;-) да надо вообще поиск на форуме придумать.

**Comment by Guria** [comment]
@zxqfox https://github.com/bem/bem-forum-content-ru/issues?q=commenter%3Atadatuta
но сделать "комменты более 10 строк", уже сложнее :)
